static/app/views/automations/components/editAutomationActions.spec.tsx

```diff
@@ -0,0 +1,62 @@
+import {AutomationFixture} from 'sentry-fixture/automations';
+import {OrganizationFixture} from 'sentry-fixture/organization';
+
+import {
+  render,
+  renderGlobalModal,
+  screen,
+  userEvent,
+  within,
+} from 'sentry-test/reactTestingLibrary';
+
+import {EditAutomationActions} from './editAutomationActions';
+
+describe('EditAutomationActions', () => {
+  it('calls delete mutation when deletion is confirmed', async () => {
+    const automation = AutomationFixture();
+    const organization = OrganizationFixture();
+
+    const mockDeleteAutomation = MockApiClient.addMockResponse({
+      url: `/organizations/${organization.slug}/workflows/${automation.id}/`,
+      method: 'DELETE',
+    });
+
+    const {router} = render(<EditAutomationActions automation={automation} />);
+    renderGlobalModal();
+
+    await userEvent.click(screen.getByRole('button', {name: 'Delete'}));
+
+    // Confirm the deletion
+    const dialog = await screen.findByRole('dialog');
+    await userEvent.click(within(dialog).getByRole('button', {name: 'Delete'}));
+
+    expect(mockDeleteAutomation).toHaveBeenCalledWith(
+      `/organizations/${organization.slug}/workflows/${automation.id}/`,
+      expect.anything()
+    );
+
+    // Redirect to the monitors list
+    expect(router.location.pathname).toBe(
+      `/organizations/${organization.slug}/issues/automations/`
+    );
+  });
+
+  it('calls update mutation when disable is selected', async () => {
+    const automation = AutomationFixture();
+    const organization = OrganizationFixture();
+
+    const mockUpdateAutomation = MockApiClient.addMockResponse({
+      url: `/organizations/${organization.slug}/workflows/${automation.id}/`,
+      method: 'PUT',
+    });
+
+    render(<EditAutomationActions automation={automation} />);
+
+    await userEvent.click(screen.getByRole('button', {name: 'Disable'}));
+
+    expect(mockUpdateAutomation).toHaveBeenCalledWith(
```

Comment:
I would add an assertion that the button now says `enable`

Reply:
I originally tried this but realized that the `automation` being passed as a prop to `EditAutomationActions` does not actually change, so the text doesn't say `enable` 😬  Is there a good way to test that in this case?

Reply:
Would moving the tests to the top-level component fix this issue? My understanding is that we would still have a single mock response for the automation GET request, which would say that the automation is still enabled. Unless you can change the mock response results after "disable" is clicked?

Reply:
Oh yes, if your component causes the query to refetch you'll need to add another `MockApiClient.addMockResponse()` before clicking the button and it should work as expected.

Though it's probably better to update the cache directly instead of refetching. That will make things happen instantly and will be simpler to test

Reply:
I think your intuition is correct that refetching is good for updating conditions. The _best_ way to do things is to update what you can in the cache and invalidate the query so a refetch happens in the background, just in case your cache updates weren't correct.